examples/island_cluster_with_vm_router/providers.tf

```diff
@@ -0,0 +1,20 @@
+terraform {
+  required_version = "~> 1.6"
+
+  required_providers {
+    google = {
+      source = "hashicorp/google"
+    }
+    google-beta = {
+      source = "hashicorp/google-beta"
+    }
+  }
+}
+
+provider "google" {
+  project = var.project_id
+}
+
+provider "google-beta" {
+  project = var.project_id
+}
```

Comment:
nit: lets remove this and provide project with resource. Also rename to `versions.tf` for consistency with other examples.

https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/f864e8add729b799529f5ede187347ed5cebd5f9/examples/safer_cluster_iap_bastion/versions.tf